src/models/room.ts

```diff
@@ -2249,6 +2288,184 @@ export class Room extends EventEmitter {
             return "Empty room";
         }
     }
+
+    /**
+     * When we receive a new visibility change event:
+     *
+     * - store this visibility change alongside the timeline, in case we
+     *   later need to apply it to an event that we haven't received yet;
+     * - if we have already received the event whose visibility has changed,
+     *   patch it to reflect the visibility change and inform listeners.
+     */
+    private applyNewVisibilityChangeEvent(event: MatrixEvent): void {
```

Comment:
from before: we should be consistent in the naming. If we're calling this an EventVisibility, then we should always use "EventVisibility".

Applies to the other functions as well.